### PR TITLE
fix: use TAG_LATEST

### PR DESCRIPTION
**Pull request by 3benbox**
Only tag the workflow docker image with "latest" when it's a merge to main.

Currently we're getting branches tagged with latest.

### CODE REVIEW

File: .github/workflows/rust-build-and-test.yml [hunk 1]
```diff
@@ -97,7 +97,13 @@ jobs:
...
           BUILD_TAG=$(echo ${{ github.sha }} | head -c 12)
         fi
-        make SHA_TAG="$BUILD_TAG" publish-docker
+        # Tag with latest if merge to main
+        if [ "${{ github.event_name }}" == "merge_group" ]; then
+          TAG_LATEST=true
+        else
+          TAG_LATEST=false
+        fi
+        make SHA_TAG="$BUILD_TAG" TAG_LATEST="$TAG_LATEST" publish-docker
```
Comment (smrz2001) [inline .github/workflows/rust-build-and-test.yml:106]: Can revert this section, actually. The `cd-to-infra.yml` workflow detects pushes to `main` and so there you could add `TAG_LATEST="true"` in the `make` command there:
```
make SHA="${{ github.SHA }}" SHA_TAG="$SHA_TAG" RELEASE_TAG="$RELEASE_TAG" TAG_LATEST="true" publish-docker
```

File: ci-scripts/publish.sh [hunk 2]
```diff
@@ -11,7 +11,6 @@
 # to login to docker. That password will be valid for 12h.
 
 docker buildx build --load -t 3box/ceramic-one .
-docker tag 3box/ceramic-one:latest public.ecr.aws/r5b3e0r5/3box/ceramic-one:latest
```
Comment (smrz2001) [inline ci-scripts/publish.sh:14]: Good catch!!!

File: .github/workflows/rust-build-and-test.yml (hunk 1 above)
Comment (3benbox) [reply .github/workflows/rust-build-and-test.yml]: Removed!

Comment (smrz2001) [reply .github/workflows/rust-build-and-test.yml]: Wait haha, I meant can keep the TAG_LATEST change in the publish script, revert changes to the test workflow yml, and then add the TAG_LATEST=true to the CD workflow.

Comment (3benbox) [reply .github/workflows/rust-build-and-test.yml:106]: Ah, updated publish to set the latest tag when a release is done.

We don't need the TAG_LATEST anymore, yeah?

Updated for that.

File: .github/workflows/cd-to-infra.yml [hunk 3]
```diff
@@ -46,6 +46,12 @@ jobs:
           # Use the release tag to deploy, if one is available.
           DEPLOY_TAG=$RELEASE_TAG
         fi
+        # Tag with latest if this is a release (not pre-release)
```
Comment (smrz2001) [inline .github/workflows/cd-to-infra.yml:49]: Can you switch this check around? Should tag `latest` only if this is a push to `main`, versus for a release.

File: .github/workflows/rust-build-and-test.yml [hunk 4]
```diff
@@ -97,7 +97,7 @@ jobs:
         else
           BUILD_TAG=$(echo ${{ github.sha }} | head -c 12)
         fi
-        make SHA_TAG="$BUILD_TAG" publish-docker
+        make SHA_TAG="$BUILD_TAG" TAG_LATEST="false" publish-docker
```
Comment (smrz2001) [inline .github/workflows/rust-build-and-test.yml:100]: Leaving this out would work as well, but being explicit might be more readable.

File: .github/workflows/cd-to-infra.yml [hunk 5]
```diff
@@ -46,6 +46,12 @@ jobs:
...
+        if [[ "${{ github.event.action }}" == "published" && "${{ github.event.release.prerelease }}" == "true" ]]; then
+          TAG_LATEST=false
+        elif [[ "${{ github.event.action }}" == "released" ]]; then
+          TAG_LATEST=true
+        fi
         make SHA="${{ github.SHA }}" SHA_TAG="$SHA_TAG" RELEASE_TAG="$RELEASE_TAG" publish-docker
```
[the remark this answers is not in the document]
Comment (3benbox) [reply .github/workflows/cd-to-infra.yml]: doh

File: .github/workflows/cd-to-infra.yml (hunk 3 above)
Comment (3benbox) [reply .github/workflows/cd-to-infra.yml:49]: updated!

File: .github/workflows/rust-build-and-test.yml (hunk 4 above)
Comment (3benbox) [reply .github/workflows/rust-build-and-test.yml:100]: I live in fear of defaults

File: .github/workflows/cd-to-infra.yml [hunk 6]
```diff
@@ -46,7 +46,13 @@ jobs:
           # Use the release tag to deploy, if one is available.
           DEPLOY_TAG=$RELEASE_TAG
         fi
-        make SHA="${{ github.SHA }}" SHA_TAG="$SHA_TAG" RELEASE_TAG="$RELEASE_TAG" publish-docker
+        # Tag with latest if this is a release (not pre-release)
```
Comment (smrz2001) [inline .github/workflows/cd-to-infra.yml:49]: This comment can be updated as well.

**Review by smrz2001 — APPROVED** [review]
Just one editorial comment otherwise LGTM! 🚀

**Pull request merged**